### PR TITLE
Add support for scheduled weight decays in RectifiedAdam.

### PR DESCRIPTION
**Pull request by leandro-gracia-gil**
RAdam implements weight decay based on AdamW, and the latter supports scheduling for both learning rate and weight decays.

This patch extends existing support of Keras schedulers for the learning rate to weight decay, matching the weight decay features of AdamW.

fixes https://github.com/tensorflow/addons/issues/1908

### CODE REVIEW

File: tensorflow_addons/optimizers/tests/rectified_adam_test.py [hunk 1]
```diff
@@ -177,3 +177,35 @@ def test_serialization():
...
+
+    assert new_optimizer.get_config()["learning_rate"] == {
+        "class_name": "ExponentialDecay",
+        "config": lr_scheduler.get_config(),
+    }
+
+    assert new_optimizer.get_config()["weight_decay"] == {
+        "class_name": "InverseTimeDecay",
+        "config": wd_scheduler.get_config(),
+    }
```
Comment (WindQAQ) [inline tensorflow_addons/optimizers/tests/rectified_adam_test.py:211]: can we also add tests about feeding `Dict` for `learning_rate` and `weight_decay`?

File: tensorflow_addons/optimizers/rectified_adam.py [hunk 2]
```diff
@@ -87,15 +87,17 @@ def __init__(
         r"""Construct a new RAdam optimizer.
 
         Args:
-            learning_rate: A `Tensor` or a floating point value. or a schedule
-                that is a `tf.keras.optimizers.schedules.LearningRateSchedule`
+            learning_rate: A `Tensor` or a floating point value, or a schedule
+                that is a `tf.keras.optimizers.schedules.LearningRateSchedule`.
```
[the remark this answers is not in the document]
Comment (leandro-gracia-gil) [reply tensorflow_addons/optimizers/rectified_adam.py:91]: The Dict is because tf.keras.optimizers.deserialize will pass a serialized config dict as learning rate or weight decay when using schedulers. It's not really intended for "regular" use. Do you have any suggestions about this?

File: tensorflow_addons/optimizers/tests/rectified_adam_test.py (hunk 1 above)
Comment (leandro-gracia-gil) [reply tensorflow_addons/optimizers/tests/rectified_adam_test.py:211]: It's already happening in the test, called by this line:
```python
new_optimizer = tf.keras.optimizers.deserialize(config)
````

In fact, adding Dict as input was done to fix a problem caught by the automatic CPU tests. What happens is that if learning rate or weight decay were keras schedulers, calling deserialize actually passes config dicts as inputs for these arguments, in which case you need to call `tf.keras.optimizers.schedules.deserialize`. That's why I had to add Dict.

File: tensorflow_addons/optimizers/rectified_adam.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (WindQAQ) [reply tensorflow_addons/optimizers/rectified_adam.py:91]: Got it, it makes sense to me.

File: tensorflow_addons/optimizers/tests/rectified_adam_test.py (hunk 1 above)
Comment (WindQAQ) [reply tensorflow_addons/optimizers/tests/rectified_adam_test.py:211]: Got it, it makes sense to me.

File: tensorflow_addons/optimizers/rectified_adam.py [hunk 3]
```diff
@@ -87,15 +87,17 @@ def __init__(
...
                 The learning rate.
             beta_1: A float value or a constant float tensor.
                 The exponential decay rate for the 1st moment estimates.
             beta_2: A float value or a constant float tensor.
                 The exponential decay rate for the 2nd moment estimates.
             epsilon: A small constant for numerical stability.
-            weight_decay: A floating point value. Weight decay for each param.
+            weight_decay: A `Tensor` or a floating point value, or a schedule
+                that is a `tf.keras.optimizers.schedules.LearningRateSchedule`.
+                Weight decay for each param.
```
[the remark this answers is not in the document]
Comment (leandro-gracia-gil) [reply tensorflow_addons/optimizers/rectified_adam.py]: Done.
